File: verticalseekbar/src/main/java/com/onoapps/hanan/verticalseekbar/SavedState.java
package com.onoapps.hanan.verticalseekbar;

import android.os.Parcel;
import android.os.Parcelable;
import android.view.View;

public class SavedState extends View.BaseSavedState {

    public float savedPercentage;
    public String savedLabel;

    public SavedState(Parcelable superState) {
        super(superState);
    }

    private SavedState(Parcel in){
        super(in);
        savedPercentage = in.readFloat();
        savedLabel = in.readString();
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        super.writeToParcel(out, flags);
        out.writeFloat(savedPercentage);
        out.writeString(savedLabel);
    }

    public static final Parcelable.Creator<SavedState> CREATOR
            = new Parcelable.Creator<SavedState>(){

        @Override
        public SavedState createFromParcel(Parcel in) {
            return new SavedState(in);
        }

        @Override
        public SavedState[] newArray(int size) {
            return new SavedState[size];
        }
    } ;
}

File: verticalseekbar/src/main/java/com/onoapps/hanan/verticalseekbar/SavedState.kt
package com.onoapps.hanan.verticalseekbar

import android.os.Parcel
import android.os.Parcelable
import android.view.View

class SavedState : View.BaseSavedState {

    var savedPercentage: Float = 0.toFloat()
    var savedLabel: String? = null

    constructor(superState: Parcelable) : super(superState) {}

    private constructor(`in`: Parcel) : super(`in`) {
        savedPercentage = `in`.readFloat()
        savedLabel = `in`.readString()
    }

    override fun writeToParcel(out: Parcel, flags: Int) {
        super.writeToParcel(out, flags)
        out.writeFloat(savedPercentage)
        out.writeString(savedLabel)
    }

    companion object {

        val CREATOR: Parcelable.Creator<SavedState> = object : Parcelable.Creator<SavedState> {

            override fun createFromParcel(`in`: Parcel): SavedState {
                return SavedState(`in`)
            }

            override fun newArray(size: Int): Array<SavedState?> {
                return arrayOfNulls(size)
            }
        }
    }
}
